File: backend/src/main/java/net/zorphy/backend/site/catan/dto/game/GameConfig.java
package net.zorphy.backend.site.catan.dto.game;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import net.zorphy.backend.site.all.http.dto.TeamDetails;
import net.zorphy.backend.site.all.http.dto.GameConfigBase;
import net.zorphy.backend.site.catan.dto.DiceConfig;
import net.zorphy.backend.site.catan.dto.enums.GameMode;

import java.util.List;

public record GameConfig(
        @NotEmpty
        @Valid
        List<TeamDetails> teams,

        @NotNull
        GameMode gameMode,

        @NotNull
        @Valid
        DiceConfig classicDice,

        @NotNull
        @Valid
        DiceConfig eventDice,

        @NotNull
        @Min(value = 1)
        Integer maxShipTurns,

        @NotNull
        @Min(value = 1)
        Integer initialShipTurns
) implements GameConfigBase {

}

File: frontend/src/app/sites/catan/dto/game/GameConfig.ts
import {DiceConfig} from "../DiceConfig";
import {Team} from "../../../../main/core/dto/Team";
import {GameMode} from "../enums/GameMode";
import {GameConfigBase} from "../../../all/dto/GameConfigBase";

export interface GameConfig extends GameConfigBase {
    teams: Team[],
    gameMode: GameMode,
    classicDice: DiceConfig,
    eventDice: DiceConfig,
    maxShipTurns: number,
    initialShipTurns: number
}